### DISCUSSION TITLE
Rationale behind ExecuteCommandWithOutputFile() and ExecuteCommandWithOutputFileTimeout()?

### DISCUSSION THREAD

**Discussion by leseb** [rook/rook#8338] [General]
While browsing the exec package (again), I'm wondering what's the purpose of both ExecuteCommandWithOutputFile() and ExecuteCommandWithOutputFileTimeout()?
It feels unnecessary to write to a file, read it and return its content when everything can be done with streaming stdout/stderr. A lot of system calls could be avoided if we just stick with `ExecuteCommand()` and co.

@travisn @BlaineEXE thoughts? Are these legacy leftovers?

**Comment by travisn** [comment]
As I recall, those were added because when the debug log level is enabled, the commands could output extra data that was not related to the desired output. It would certainly be nice to consolidate these methods, but we would have to be careful about the debug logging.

**Reply by leseb** [reply]
Which debug level? Client CLI debug? 🤔

**Reply by travisn** [reply]
Maybe it's not a common issue anymore. We used to increase the `debug` log level for some ceph components when the rook operator log level was set to DEBUG, but I don't think this is done anymore. What still is likely an issue is how the commands would be affected if someone increases the log level as part of the ceph.conf overrides.

**Reply by leseb** [reply]
Yes increasing the client level debug would certainly mess up the ceph commands output but the ceph.conf file from the operator is not configurable AFAIK so we shouldn't hit that issue right? The ceph.conf overrides only affect daemon's pod IIRC.

**Reply by travisn** [reply]
Yes the operator should be picking up the ceph.conf overrides, it combines into the ceph.conf generated under /var/lib/rook/<namespace>

**Reply by leseb** [reply]
Ah right, ok we could avoid applying the debug settings though if it's just for that :)

**Reply by travisn** [reply]
Do you mean that we could skip applying the debug settings in the operator even if they are specified? I don't know why they would want to increase the ceph debug levels in the operator, so that might be worth it if we can simplify these exec helpers.

**Reply by leseb** [reply]
I don't know why would anybody do this, but it'd be nice to simplify the exec helpers for sure, I'll give it a stab.
